src/cargo/ops/cargo_check.rs

```diff
@@ -0,0 +1,13 @@
+use core::Workspace;
+use util::important_paths::find_root_manifest_for_wd;
+use util::{CliResult, Config};
+
+pub fn with_check_ws<F>(flag_manifest_path: Option<String>,
+                        config: &Config, f: F)
+                        -> CliResult<Option<()>>
+    where F: FnOnce(&Workspace) -> CliResult<Option<()>>
+{
+    let root = find_root_manifest_for_wd(flag_manifest_path, config.cwd())?;
+    let ws = Workspace::new(&root, config)?;
```

Comment:
I think there was a borrow check issue with doing that - `ws` has to have a shorter lifetime than `root`. I agree it is kind of a small function, but there is already a little bit of code that needs to be inlined (to make the config) and it seemed like voodoo from the client's perspective - there seems no reason to allow the flexibility - so I thought it better to present a simpler API